### PR TITLE
checking header types

### PR DESCRIPTION
**Pull request by rf**
@Raynos

### CODE REVIEW

**Comment by Raynos** [comment]
Lgtm. r: @jcorbin 

This manual check gives a better error then the underlying `bufrw` error. Especially reporting the name of the header that is wrong helps debugging the issue.

File: node/errors.js [hunk 1]
```diff
@@ -164,6 +164,14 @@ module.exports.InvalidHandlerForRegister = TypedError({
     handler: null
 });
 
+module.exports.InvalidHeaderTypeError = TypedError({
+    type: 'tchannel.invalid-header-type',
+    message: 'invalid header type for header {name}; expected string, got' +
```
Comment (jcorbin) [inline node/errors.js]: we don't do line length limit in tchannel, please combine this

File: node/v2/header.js [hunk 2]
```diff
@@ -78,6 +78,13 @@ HeaderRW.prototype.writeInto = function writeInto(headers, buffer, offset) {
         if (res.err) return res;
         offset = res.offset;
 
+        if (typeof headers[key] !== 'string') {
+            return errors.InvalidHeaderTypeError({
```
Comment (jcorbin) [inline node/v2/header.js]: erm, this needs to return a result... so `return WriteResult.error(errors...);`

Comment (jcorbin) [inline node/v2/header.js]: also, shouldn't a buffer be valid too, not just a string?

Comment (jcorbin) [inline node/v2/header.js]: Also, github won't let me comment on it, but you want to do the same check in `HeaderRW#byteLength` above circa L57; there it'll be `LengthResult.error(....)`.

Finally, the fact that any of this didn't fail a test is a bug, please cover these paths and prove they work.

**Comment by Raynos** [comment]
Let's adding a failing test too.

**Comment by rf** [comment]
:+1:

**Pull request merged**